src/Messaging/NBB.Messaging.Rx/MessagingObservable.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NBB.Messaging.Abstractions;
+
+namespace NBB.Messaging.Rx
+{
+    public class MessagingObservable<TMessage> : IMessagingObservable<TMessage>
+    {
+        private readonly IDisposable _subscription;
+        private readonly ConcurrentDictionary<int, ObserverSubscription> _subscriptions = new();
+
+        public MessagingObservable(IMessageBusSubscriber messageBusSubscriber)
+        {
+            //TODO improve async
+            _subscription = messageBusSubscriber.SubscribeAsync<TMessage>(InvokeObservers).GetAwaiter().GetResult();
```

Comment:
This messaging subscription will consume messages even if no observers are subscribed yet. This way some messages will be lost. This is ok for a "hot observable" but is this acceptable for our scenarios? 

Here is a less flexible approach where the messaging subscription and observable subscription are coupled to avoid message loss: https://github.com/danielwertheim/mynatsclient/tree/master#streamsubscribe-vs-streamsubscribesafe.